Add slot support for components

awinny opened the pull request:
**Describe pull-request**</br>
- Added slots to header, footer and navigation
- Added theme support for components missing it
- Added header state to the store
- Improved the store handling
- Changed name for the theme handling

awinny committed:
Add slot support forcomponents
- Added slots to header, footer and navigation
- Added theme support for components missing it
- Added header state to the store
- Improved the store handling
- Changed name for the theme handling

helloimela commented on src/components/navigation/navigation.tsx:
```diff
@@ -66,6 +69,8 @@ export class Navigation {
               </li>
             )}
+
+            <slot name="secondary-items" />
```
Is the slot should be outside `<ul>` ?  If we have it inside `<ul>`, it's limiting us to ALWAYS add `<li>` before any component we wanted to put there by ourselves. And more documentation needed to make it clear as well. So I suggest to have it outside `<ul>` so people can add whatever component they have in the header or navigation. For example if we have react component called `<my-dropdown>`, then it will be weird if it is rendered as:
`<ul><my-dropdown /></ul>`

awinny replied:
I agreed, have changed markup to provide a more fitting structure.

awinny committed:
Changed component template setup
- Changed header, footer and navigation item container from ul to nav
- Adjusted styling for nav items

awinny committed:
Fix for nested style rules
- Fixed incorrect nested scss rules
